helpers/mysql/lib/opentelemetry/helpers/mysql.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0module OpenTelemetry
+
+module OpenTelemetry
+  module Helpers
+    # This class contains helpers for MySQL instrumentation libraries, like mysql2 and
+    # trilogy. It is intended for use by instrumentation developers and not to
+    # be called directly by an application.
+    #
+    # To use this in your instrumentation, the `Instrumentation` class for
+    # your gem must contain configuration options for:
+    #  * `:span_name`
+    #  Example:
+    #  `option :span_name, default: :statement_type, validate: %I[statement_type db_name db_operation_and_name]`
+    #
+    # @api public
+    class MySQL
+      QUERY_NAMES = [
+        'set names',
+        'select',
+        'insert',
+        'update',
+        'delete',
+        'begin',
+        'commit',
+        'rollback',
+        'savepoint',
+        'release savepoint',
+        'explain',
+        'drop database',
+        'drop table',
+        'create database',
+        'create table'
+      ].freeze
+
+      QUERY_NAME_REGEX = Regexp.new("^(#{QUERY_NAMES.join('|')})", Regexp::IGNORECASE)
```

Comment:
There's a frustrating corner case with this regex that would be good to fix (doesn't have to be done here - I'll do it if you don't): this regex assumes queries start with the operation, and don't contain preceding whitespace or comments. In a lot of our production code, this assumption is wrong (we have "marginalia" comments that contain useful telemetry that'll be captured in slow query logs, for example). Consequently, most of our MySQL spans are simply named `mysql` 😞.

Reply:
That sounds very frustrating! I have a potential fix here: c694e556d67b5ad02d7a9604d11bc9d2c5284472 

Now, the statement type will be set to the first matched item in `QUERY_NAMES`, regardless of where it appears in the SQL string. I used a marginalia comment to test the change.